server/simple_server.go

```diff
@@ -283,3 +305,34 @@ func GetIP(r *http.Request) (string, error) {
 	}
 	return userIP.String(), nil
 }
+
+// ContextKey used to create context keys
+type ContextKey int
+
+// UserIPKey is key to set/retrieve value from context
+const UserIPKey contextKey = 0
```

Comment:
Mind tossing the consts in a const ( ) block? Just visually groups them together a bit more.